### PR TITLE
[lit-next] Fixes `@query` with cache setting when Babel compilation is used

### LINKED ISSUE #1584: [lit-next] Enable babel decorator testing
## Description
Babel decorator testing is current disabled. It should be re-enabled.

## Acceptance criteria
PR landed with testing enabled and passing.

### PR DESCRIPTION
**Pull request by sorvell**
Fixes #1584, #1572.

This PR also re-enables testing under Babel compilation. This is now done in ReactiveElement and not LitElement.

### CODE REVIEW

File: packages/reactive-element/package.json [hunk 1]
```diff
@@ -39,10 +39,11 @@
     }
   },
   "scripts": {
-    "build": "npm run clean && tsc --build && rollup -c",
+    "build": "npm run clean && tsc --build && rollup -c && npm run build:babel",
```
Comment (justinfagnani) [inline packages/reactive-element/package.json:42]: Can you all call out to `build:ts` for consistency?

File: packages/reactive-element/src/test/decorators/property_test.ts [hunk 2]
```diff
@@ -264,7 +264,13 @@ suite('@property', () => {
         return {foo: {type: Number, reflect: true}};
       }
 
-      foo = 1;
+      declare foo: number;
```
Comment (justinfagnani) [inline packages/reactive-element/src/test/decorators/property_test.ts:267]: Have we now lost all coverage of declared class fields in TS with `useDefineForClassFields=false`

**Review by justinfagnani — COMMENTED** [review]
LG except a concern about testing declared fields in TS.

**Review by justinfagnani — APPROVED** [review]

**Comment by sorvell** [comment]
Merging with IE test timeouts since those will be fixed separately.

**Pull request merged**